Lanchonete/LanchoneteTests/Managers/SandwichManagerTests.swift

```diff
@@ -0,0 +1,151 @@
+    
+    func test_GIVEN_a_menu_sandwich_WHEN_I_remove_an_ingredient_THEN_the_price_is_updated(){
+        //Given
+        let sandwichManager = SandwichManager(with: .xBacon)
+        
+        //When
+        sandwichManager.removeIngredient(.bacon)
+        
+        //Then
+        XCTAssertEqual(sandwichManager.getPrice(), 4.5)
+    }
+    
+    func test_GIVEN_a_menu_sandwich_WHEN_I_remove_an_ingredient_THEN_the_ingredient_list_is_updated(){
+        //Given
+        let sandwichManager = SandwichManager(with: .xBacon)
+
+        //When
+        sandwichManager.removeIngredient(.bacon)
+
+        let expectedResult = [Ingredient.hamburguer, .cheese]
+        
+        //Then
+        XCTAssertEqual(sandwichManager.getIngredients(),expectedResult)
+    }
+    
+    func test_GIVEN_a_menu_sandwich_WHEN_I_remove_an_ingredient_that_does_not_exist_THEN_the_price_should_be_the_same(){
+        //Given
+        let sandwichManager = SandwichManager(with: .xBacon)
+        
+        //When
+        sandwichManager.removeIngredient(.egg)
+        
+        //Then
+        XCTAssertEqual(sandwichManager.getPrice(), 7.0)
+    }
+    
+    func test_GIVEN_a_menu_sandwich_WHEN_I_remove_an_ingredient_that_does_not_exist_THEN_the_ingredient_list_should_be_the_same(){
+        //Given
+        let sandwichManager = SandwichManager(with: .xBacon)
+
+        //When
+        sandwichManager.removeIngredient(.egg)
+
+        let expectedResult = [Ingredient.hamburguer, .cheese, .bacon]
+        
+        //Then
+        XCTAssertEqual(sandwichManager.getIngredients(),expectedResult)
+    }
+    
+    func test_GIVEN_a_customized_sandwich_WHEN_I_add_ingredient_THEN_the_price_is_updated(){
+        //Given
+        let sandwichManager = SandwichManager()
+        
+        //When
+        sandwichManager.addIngredient(.egg)
+        sandwichManager.addIngredient(.egg)
+        sandwichManager.addIngredient(.cheese)
+        
+        //Then
+        XCTAssertEqual(sandwichManager.getPrice(), 5.5)
+    }
+    
+    func test_GIVEN_a_customized_sandwich_WHEN_I_add_duplicated_ingredients_and_remove_one_THEN_the_price_is_updated(){
+        //Given
+        let sandwichManager = SandwichManager()
+        
+        //When
+        sandwichManager.addIngredient(.egg)
+        sandwichManager.addIngredient(.egg)
+        sandwichManager.addIngredient(.bacon)
+        
+        sandwichManager.removeIngredient(.egg)
+        
+        //Then
+        XCTAssertEqual(sandwichManager.getPrice(), 4.5)
+    }
+    
+    func test_GIVEN_a_customized_sandwich_WHEN_I_add_multiple_ingredients_and_remove_them_all_THEN_the_price_should_be_0(){
+        //Given
+        let sandwichManager = SandwichManager()
+        
+        //When
+        sandwichManager.addIngredient(.egg)
+        sandwichManager.addIngredient(.egg)
+        sandwichManager.addIngredient(.bacon)
+        sandwichManager.addIngredient(.bacon)
+        
+        sandwichManager.removeIngredient(.egg)
```

Comment:
This kind of tests is a bit tricky, because it's not independent.
You need two actions to assert something (addIngredient and removeIngredient).
You have a way to initialize your manager with a state, right? As in: a list of ingredients already setup? Using that, you could test the removeIngredient in an isolated way. And the `setup` is also tested in isolation in another test. It's just a thought, not a request

Reply:
I think I can create a removeAll ingredients but I believe it's not necessary to create another constructor just for the tests.